apps/framework-cli/src/cli/routines/peek.rs

```diff
@@ -0,0 +1,96 @@
+use crate::cli::display::Message;
+use crate::infrastructure::olap::clickhouse::mapper::std_table_to_clickhouse_table;
+use crate::infrastructure::olap::clickhouse_alt_client::{
+    get_pool, retrieve_infrastructure_map, select_some_as_json,
+};
+use crate::project::Project;
+
+use super::{RoutineFailure, RoutineSuccess};
+
+use futures::StreamExt;
+use std::sync::Arc;
+
+pub async fn peek(
+    project: Arc<Project>,
+    data_model_name: String,
+    limit: u8,
+) -> Result<RoutineSuccess, RoutineFailure> {
+    let pool = get_pool(&project.clickhouse_config);
+    let mut client = pool.get_handle().await.map_err(|_| {
+        RoutineFailure::error(Message::new(
+            "Failed".to_string(),
+            "Error connecting to storage".to_string(),
+        ))
+    })?;
+
+    let infra = retrieve_infrastructure_map(&mut client, &project.clickhouse_config)
+        .await
+        .map_err(|_| {
+            RoutineFailure::error(Message::new(
+                "Failed".to_string(),
+                "Error retrieving current state".to_string(),
+            ))
+        })?
+        .ok_or_else(|| {
+            RoutineFailure::error(Message::new(
+                "Failed".to_string(),
+                "No state found".to_string(),
+            ))
+        })?;
+
+    let table = infra
+        .tables
+        .iter()
+        .find(|(key, _)| key.starts_with(&data_model_name))
+        .map(|(_, table)| table)
```

Comment:
Maybe we should try exact match before searching by prefix.
Probably do some logging/printing like "{} not found, showing {}".
If we try to be helpful we should probably add case insensitivity and ignore underscores.

BTW `find_map`

Reply:
to do exact match in the infra map, suffix the name with version twice (weird, i know)

https://github.com/514-labs/moose/pull/1877/files#diff-67db23e6b214a8a103adf10ce2ad885081921452099eaee572ccd491f65b4241R64